src/deepsparse/pipeline.py

```diff
@@ -142,13 +145,23 @@ def __init__(
         alias: Optional[str] = None,
         context: Optional[Context] = None,
         executor: Optional[Union[ThreadPoolExecutor, int]] = None,
+        logger: Optional[BaseLogger] = None,
     ):
         self._model_path_orig = model_path
         self._model_path = model_path
         self._engine_type = engine_type
         self._batch_size = batch_size
         self._alias = alias
         self.context = context
+        self.logger = logger
+
+        if self.logger and not self._alias:
```

Comment:
My philosophy here is: is somebody wants to enable logging, they should be explicit in the naming of their pipelines. Feel free to challenge this opinion.

Reply:
Disagreed here, let's just default to using the task name - the server will always handle alias and any other user using a logger shouldn't have to worry about setting one unless dealing with multiple models